### PR TITLE
feat: Add CalcualteFee in GRPC

### LINKED ISSUE #590: No fields for transaction Fee
## Describe the problem you'd like to have solved

Cannot find `transaction fee` details in API or gRPC Call

### Describe the ideal solution

We need to create/add API/fee to calculate the total transaction fee.

### PR DESCRIPTION
**Pull request by ZigBalthazar**
## Description

* add `calculateFee` to GRPC
* change wallet `setFee `and `calcFee`
* change some typo
*  debug `./makefile`

## Related issue(s)

- Fixes #590

### CODE REVIEW

File: node/node.go [hunk 1]
```diff
@@ -42,7 +42,7 @@ func NewNode(genDoc *genesis.Genesis, conf *config.Config,
 	// Initialize the logger
 	logger.InitLogger(conf.Logger)
 
-	logger.Info("You are running a pactus Blockchain",
+	logger.Info("You are running a pactus block chain",
```
Comment (b00f) [inline node/node.go:45]: Please revert this change.

File: state/facade.go [hunk 2]
```diff
@@ -49,4 +50,5 @@ type Facade interface {
 	Params() param.Params
 	Close() error
 	Fingerprint() string
+	CalcFee(amount int64, payloadType payload.Type) (int64, error)
```
Comment (b00f) [inline state/facade.go:53]: To be consistent with other part of the project, rename it to `CalculateFee`

File: state/mock.go [hunk 3]
```diff
@@ -209,3 +210,7 @@ func (m *MockState) AddPendingTxAndBroadcast(trx *tx.Tx) error {
 func (m *MockState) Params() param.Params {
 	return m.TestParams
 }
+
+func (m *MockState) CalcFee(_ int64, payloadType payload.Type) (int64, error) {
+	return 0, nil
```
Comment (b00f) [inline state/mock.go:215]: Let's return random number. Like: `ts.RandInt64(1^e9)`
Maybe some test fails, (Unbond Tests?). Then check type here.

File: state/state_test.go [hunk 4]
```diff
@@ -728,3 +729,37 @@ func TestCommittingInvalidBlock(t *testing.T) {
...
+		{1 * 1e9, 1, 100000, errors.ErrInvalidFee},
+		{1 * 1e9, 100001, 100000, errors.ErrInvalidFee},
+		{1 * 1e9, 100000, 100000, errors.ErrNone},
+
+		{1 * 1e12, 1, 1000000, errors.ErrInvalidFee},
+		{1 * 1e12, 1000001, 1000000, errors.ErrInvalidFee},
+		{1 * 1e12, 1000000, 1000000, errors.ErrNone},
+	}
+	for _, test := range tests {
+		fee, err := td.state2.CalcFee(int64(test.amount), payload.PayloadTypeTransfer)
```
Comment (b00f) [inline state/state_test.go:754]:
```suggestion
		fee, err := td.state2.CalcFee(test.amount, payload.PayloadTypeTransfer)
```

This fix the linting issue

File: state/state_test.go [hunk 5]
```diff
@@ -728,3 +729,37 @@ func TestCommittingInvalidBlock(t *testing.T) {
...
+		assert.NoError(t, err)
+		assert.Equal(t, int64(test.expectedFee), fee)
```
Comment (b00f) [inline state/state_test.go:756]:
```suggestion
		assert.Equal(t, test.expectedFee, fee)
```

Same.

File: wallet/client_test.go [hunk 6]
```diff
@@ -77,6 +77,18 @@ func (s *transactionServer) GetTransaction(_ context.Context,
 	return nil, nil
 }
 
+func (s *transactionServer) CalculateFee(_ context.Context,
+	_ *pactus.CalculateFeeRequest) (*pactus.CalculateFeeResponse, error) {
+
+	return &pactus.CalculateFeeResponse{Fee: 0}, nil
+}
+
+func (s *transactionServer) GetFee(_ context.Context,
```
Comment (b00f) [inline wallet/client_test.go:86]: This function is not necessary here.

File: wallet/client.go [hunk 7]
```diff
@@ -84,3 +86,11 @@ func (c *grpcClient) getTransaction(id tx.ID) (*pactus.GetTransactionResponse, e
 
 	return res, nil
 }
+
+func (c *grpcClient) getFee(amount int64, payloadType payload.Type) (int64, error) {
+	res, err := c.transactionClient.CalculateFee(context.Background(), &pactus.CalculateFeeRequest{Amount: amount, PayloadType: pactus.PayloadType(payloadType)})
```
Comment (b00f) [inline wallet/client.go:91]:
```suggestion
	res, err := c.transactionClient.CalculateFee(context.Background(), &pactus.CalculateFeeRequest{
	Amount: amount, PayloadType: pactus.PayloadType(payloadType)})
```

Break this long line

**Review by b00f — APPROVED** [review]

**Comment by b00f** [comment]
@ZigBalthazar 

Thanks for this PR. It fix the `CalculateFee` related issues.

Still, some items need to be fixed:
1- Please update [grpc-gateway.config.yaml](https://github.com/pactus-project/pactus/blob/main/www/grpc/grpc-gateway.config.yaml)
2- Make sure the linting check passes: `make check`
3- Fix the GUI issues.

**Comment by ZigBalthazar** [comment]
@b00f

File: cmd/gtk/utils.go [hunk 8]
```diff
@@ -181,7 +182,11 @@ func updateFeeHint(lbl *gtk.Label, amtStr string, w *wallet.Wallet) {
 	if err != nil {
 		updateHintLabel(lbl, "")
 	} else {
-		fee := w.CalculateFee(amount)
+		fee, err := w.CalculateFee(amount, payload.PayloadTypeTransfer)
```
Comment (b00f) [inline cmd/gtk/utils.go:185]: This only works for Transfer type.
To ensure the system works fine with Non-transfer payloads, it's better to pass the payload type as an argument to this function.

File: node/node.go [hunk 9]
```diff
@@ -42,7 +42,7 @@ func NewNode(genDoc *genesis.Genesis, conf *config.Config,
 	// Initialize the logger
 	logger.InitLogger(conf.Logger)
 
-	logger.Info("You are running a pactus Blockchain",
+	logger.Info("You are running a pactus blockchain",
```
Comment (b00f) [inline node/node.go:45]:
```suggestion
	logger.Info("You are running a Pactus blockchain",
```

File: state/state_test.go [hunk 10]
```diff
@@ -728,3 +729,37 @@ func TestCommittingInvalidBlock(t *testing.T) {
...
+		{1 * 1e9, 1, 100000, errors.ErrInvalidFee},
+		{1 * 1e9, 100001, 100000, errors.ErrInvalidFee},
+		{1 * 1e9, 100000, 100000, errors.ErrNone},
+
+		{1 * 1e12, 1, 1000000, errors.ErrInvalidFee},
+		{1 * 1e12, 1000001, 1000000, errors.ErrInvalidFee},
+		{1 * 1e12, 1000000, 1000000, errors.ErrNone},
+	}
+	for _, test := range tests {
+		fee, err := td.state2.CalculateFee(test.amount, payload.PayloadTypeTransfer)
```
Comment (b00f) [inline state/state_test.go:754]: Let's make this test better. Change tests structure to:
```
tests := []struct {
		amount          int64
		pldType         payload.Type
		fee             int64
		expectedFee     int64
		expectedErrCode int
	}
	```
	
	You can add more tests for non-transfer types

**Review by b00f — CHANGES_REQUESTED** [review]

**Comment by b00f** [comment]
@ZigBalthazar 
Thanks. Everything looks good to me. Just 2 small changes.

File: cmd/gtk/dialog_transaction_bond.go [hunk 11]
```diff
@@ -54,7 +55,7 @@ func broadcastTransactionBond(wallet *wallet.Wallet, valAddrs []crypto.Address)
 
 	onAmountChanged := func() {
 		amtStr, _ := amountEntry.GetText()
-		updateFeeHint(amountHint, amtStr, wallet)
+		updateFeeHint(amountHint, amtStr, wallet, payload.PayloadTypeBond)
```
Comment (kehiy) [inline cmd/gtk/dialog_transaction_bond.go:58]:
```suggestion
		updateFeeHint(amountHint, amtStr, wallet,  payload.PayloadTypeBond)
```

**Review by b00f — APPROVED** [review]

**Pull request merged**